addon/mixins/run.js

```diff
@@ -449,9 +449,9 @@ function cancelTimer(cancelId) {
   run.cancel(cancelId);
 }
 
-function cancelPoll(label) {
-  pollTaskLabels[label] = undefined;
-  queuedPollTasks[label] = undefined;
+function cancelPoll(token) {
+  pollTaskTokens[token] = undefined;
```

Comment:
i think we may want to `delete` the properties, otherwise the `pollTaskToken` and `queuePollTask` property list will still grow unbounded.

Reply:
When you say delete, do you mean splice out the item?

Reply:
ah this is an array not an object, strange... But ya, if an array is appropriate then we should consider splicing/replacing.

Is there a reason this is an array and not an Object?

Reply:
Actually you were correct in the first place. It's a dictionary. We should delete the prop. I can do that.